src/elements/unified-share-modal-element/types.js

```diff
@@ -0,0 +1,82 @@
+// @flow
+import type { ItemType } from '../../common/types/core';
+import type { item, sharedLinkType } from '../../features/unified-share-modal/flowTypes';
+
+// "SLS" denotes values that are used in the Shared Link Settings modal
+type USMElementEnterpriseDataType = {
+    enterpriseName: string,
+    serverURL: string, // SLS
+};
+
+export type USMElementUserDataType = {
+    id: string,
+    userEnterpriseData: USMElementEnterpriseDataType,
+};
+
+export type USMElementSharedLinkType =
+    | { canInvite: boolean }
+    | (sharedLinkType &
+          USMElementEnterpriseDataType & {
+              canChangeDownload: boolean, // SLS
+              canChangePassword: boolean, // SLS
+              canChangeVanityName: boolean, // SLS
+              canInvite: boolean,
+              directLink: string, // SLS
+              isDirectLinkAvailable: boolean, // SLS
+              isDownloadAvailable: boolean, // SLS
+              isDownloadEnabled: boolean, // SLS
+              isPasswordAvailable: boolean, // SLS
+              isPasswordEnabled: boolean, // SLS
+              vanityName: string,
+          });
+
+export type USMElementItemDataType = {
+    item: item,
+    sharedLink: USMElementSharedLinkType,
+};
+
+export type USMElementItemAPIResponse = {
+    description: string,
+    etag: string,
+    extension: string,
+    id: string,
+    name: string,
+    permissions: {
+        can_annotate: boolean,
+        can_comment: boolean,
+        can_delete: boolean,
+        can_download: boolean,
+        can_edit?: boolean,
```

Comment:
any reason why this one is optional? (cannot have this field on a folder item type?)

Reply:
Good catch. It turns out that `can_edit` is only available for annotations and comments, so I updated the Item API requests to include the `allowed_invitee_roles` field. The normalizer function now checks that array for the presence of the "editor" role.